app/api/constituency_lookup/[postcode]/[[...address]]/route.ts

```diff
@@ -8,6 +8,56 @@ import { validatePostcode, normalizePostcode } from "@/utils/Postcodes";
 // Force using 'nodejs' rather than 'edge' - edge won't have the filesystem containing SQLite
 export const runtime = "nodejs";
 
+type DCData = {
+  boundary_changes?: {
+    current_constituencies_official_identifier: string;
+    current_constituencies_name: string;
+    new_constituencies_official_identifier: string;
+    new_constituencies_name: string;
+    CHANGE_TYPE: string;
+  };
+  addresses?: {
+    address: string;
+    postcode: string;
+    slug: string;
+    url: string;
+  }[];
+};
+
+const dc_base_url = "https://developers.democracyclub.org.uk/api/v1/";
+const dc_params = `/?auth_token=${process.env.DC_API_KEY}&parl_boundaries=1`;
+
+async function fetch_dc_api(
+  postcode: string,
+  addressSlug?: string,
+): Promise<DCData | null> {
+  const dc_url =
+    dc_base_url +
+    (addressSlug ? "address/" + addressSlug : "postcode/" + postcode) +
+    dc_params;
```

Comment:
Use `URLSearchParams` for this to tidy it up?

See: https://stackoverflow.com/a/58437909

Reply:
Yeah makes sense I've kept it outside the function so it gets re-used between executions.